### PR TITLE
feat(cardano-services): run multiple services from a single HTTP server

### PR DESCRIPTION
**Pull request by rhyslbw**
# Context
The current design implements a HTTP server per service, which makes the design modular, but makes deployment and the client-side a bit cumbersome which is resolved by attaching all services to a single server under a unique path.

# Proposed Solution
- adds `HttpService` abstract class.
- refactors `HttpServer` to a concrete class, with services as dependencies.
- N.B. the cli and run entrypoints will be hoisted to the top level in a later commit.

# Important Changes Introduced

### CODE REVIEW

File: packages/cardano-services/src/StakePoolSearch/StakePoolSearchHttpServer.ts [hunk 1]
```diff
@@ -1,53 +1,40 @@
 import { Cardano, ProviderError, ProviderFailure, StakePoolQueryOptions } from '@cardano-sdk/core';
```
Comment (rhyslbw) [inline packages/cardano-services/src/StakePoolSearch/StakePoolSearchHttpServer.ts:1]: This file will be renamed to `StakePoolHttpService.ts` before merge.

File: packages/cardano-services/src/TxSubmit/TxSubmitHttpServer.ts [hunk 2]
```diff
@@ -1,68 +1,57 @@
 import { Cardano, ProviderError, ProviderFailure, TxSubmitProvider } from '@cardano-sdk/core';
```
Comment (rhyslbw) [inline packages/cardano-services/src/TxSubmit/TxSubmitHttpServer.ts:1]: This file will be renamed to `TxSubmitHttpService.ts` before merge.

**Review by mkazlauskas — COMMENTED** [review]
Looking good!

File: packages/cardano-services/src/Http/HttpServer.ts [hunk 3]
```diff
@@ -25,59 +26,46 @@ export type HttpServerConfig = {
...
 
-export abstract class HttpServer extends RunnableModule {
+export class HttpServer extends RunnableModule {
   public app: express.Application;
   public server: http.Server;
+  #config: HttpServerConfig;
   #dependencies: HttpServerDependencies;
 
-  protected constructor(public config: HttpServerConfig, { logger = dummyLogger, ...rest }: HttpServerDependencies) {
+  constructor(public config: HttpServerConfig, { logger = dummyLogger, ...rest }: HttpServerDependencies) {
```
[the remark this answers is not in the document]
Comment (rhyslbw) [reply packages/cardano-services/src/Http/HttpServer.ts]: Good catch. Resolved in 364003037d9d351e1b5ee8d36f248a6fda0dc76d

File: packages/cardano-services/src/Http/HttpService.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import { HealthCheckResponse, ProviderError } from '@cardano-sdk/core';
+import { Logger, dummyLogger } from 'ts-log';
+import express from 'express';
+
+export interface HttpServiceDependencies {
+  healthCheck: () => Promise<HealthCheckResponse>;
```
[the remark this answers is not in the document]
Comment (rhyslbw) [reply packages/cardano-services/src/Http/HttpService.ts]: I went with _abstract_ in dc87a3a594e77c63711c283d124e6bf6af20b117

**Review by mkazlauskas — APPROVED** [review]
Nice work! The concrete http provider service looks great, very clean :+1:

**Review by mkazlauskas — APPROVED** [review]

**Pull request merged**